schema/schema.go

```diff
@@ -0,0 +1,102 @@
+package schema
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/cloudfoundry/cli/cf/terminal"
+)
+
+type SchemaParser struct {
+	ui terminal.UI
+}
+
+func NewSchemaParser(userInterface terminal.UI) *SchemaParser {
+	return &SchemaParser{
+		ui: userInterface,
+	}
+}
+
+func (p *SchemaParser) ParseSchema(schema string) (string, error) {
+	var jsontype map[string]interface{}
+	json.Unmarshal([]byte(schema), &jsontype)
+	res, err := p.parseObject("", jsontype)
+	if err != nil {
+		return "", err
+	}
+	jsonResult, err := json.Marshal(res)
+	if err != nil {
+		return "", err
+	}
+	return string(jsonResult), nil
+}
+
+func (p *SchemaParser) parseObject(key string, input map[string]interface{}) (map[string]interface{}, error) {
+	var result map[string]interface{} = make(map[string]interface{})
+
+	if prop, ok := input["properties"]; ok {
+		for k, v := range prop.(map[string]interface{}) {
```

Comment:
Would the code be clearer if these variables have longer names (e.g. `key`, `value`)? (`type` is a keyword, though, unfortunately…)